tests/page-objects/default/enketo/pregnancy.wdio.page.js

```diff
@@ -1,201 +1,72 @@
-};
-const riskFactors = (value) => {
-  return $$(`${FORM} input[name="/pregnancy/risk_factors/risk_factors_present/${value}_condition"]`);
-};
-const patientNameSummary = () => $(`${SUMMARY_SECTION} span[data-value=" /pregnancy/patient_name "]`);
-const weeksPregnantSummary = () => $(`${SUMMARY_SECTION} span[data-value=" /pregnancy/weeks_since_lmp_rounded "]`);
-const eddSummary = () => $(`${SUMMARY_SECTION} span[data-value=" /pregnancy/summary/edd_summary "]`);
-const riskFactorsSummary = () => {
-  return $$(SUMMARY_SECTION +
-    ' :not(label.disabled):not(label.or-appearance-h3) span[data-itext-id*="/pregnancy/summary/r_risk"]');
-};
-const dangerSignsSummary = () => {
-  return $$(`${SUMMARY_SECTION} :not(label.disabled) span[data-itext-id*="/pregnancy/summary/r_danger_sign_"]`);
-};
-
-const selectGestationAge = async (value = GESTATION_AGE.edd) => {
-  const getAge = await gestationalAge(value);
-  await getAge.waitForDisplayed();
-  await getAge.click();
-};
-
-const setDeliveryDate = async (value = moment().add(1, 'month').format('YYYY-MM-DD')) => {
-  const date = await deliveryDate();
-  await date.waitForDisplayed();
-  await date.setValue(value);
-  await (await $('#form-title')).click(); // defocus from date
-};
-
-const getConfirmationDetails = async () => {
-  return {
-    eddConfirm: await (await eddConfirmation()).getText(),
-    weeksPregnantConfirm: await (await weeksPregnantConfirmation()).getText(),
-  };
-};
-
-const setANCVisitsPast = async (value = 0) => {
-  const visits = await ancVisitsPast();
-  await visits.waitForDisplayed();
-  await visits.setValue(value);
-};
+const commonEnketoPage = require('@page-objects/default/enketo/common-enketo.wdio.page');
 
 const setFutureVisitDate = async (value = moment().add(1, 'day').format('YYYY-MM-DD')) => {
-  const date = await futureVisitDate();
+  const date = await $('section[name="/pregnancy/anc_visits_hf/anc_visits_hf_next/anc_next_visit_date"] ' +
+    'input.ignore.input-small');
   await date.waitForDisplayed();
   await date.setValue(value);
 };
 
-// The selector for the 'riskFactors' is different depending on the option selected in the 'first pregnancy' question
-const selectAllRiskFactors = async (value = FIRST_PREGNANCY_VALUE.no) => {
-  const cbRisks = await riskFactors(value);
-  for (const risk of cbRisks) {
-    if (await risk.getAttribute('value') !== 'none') {
-      await risk.click();
-    }
-  }
-  return cbRisks.length - 1; // Subtract 1 due to the 'none' option
-};
-
-const getSummaryDetails = async () => {
-  return {
-    patientNameSumm: await patientNameSummary().getText(),
-    weeksPregnantSumm: await weeksPregnantSummary().getText(),
-    eddSumm: await eddSummary().getText(),
-    riskFactorsSumm: await riskFactorsSummary().length,
-    dangerSignsSumm: await dangerSignsSummary().length,
-  };
-};
+const submitDefaultPregnancy = async () => {
```

Comment:
Adding a flag to not submit the form can make this function more reusable. This would allow us to use this code in multiple tests that require validation of the summary report before submitting the default pregnancy form. What do you think?

Reply:
That is a really good idea, I will add the flag.
Thank you! 🙂